src/test/java/org/inferred/freebuilder/processor/NullablePropertyFactoryTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.inferred.freebuilder.processor;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import static org.mockito.Answers.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+
+import org.inferred.freebuilder.processor.Metadata.Property;
+import org.inferred.freebuilder.processor.NullablePropertyFactory.CodeGenerator;
+import org.inferred.freebuilder.processor.PropertyCodeGenerator.Config;
+import org.inferred.freebuilder.processor.util.testing.ModelRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import javax.annotation.Nullable;
+import javax.lang.model.element.ExecutableElement;
+
+/** Unit tests for {@link Analyser}. */
+@RunWith(MockitoJUnitRunner.class)
+public class NullablePropertyFactoryTest {
+
+  @Rule public final ModelRule model = new ModelRule();
+  @Mock(answer = RETURNS_SMART_NULLS) private Config config;
+  private final Property property = new Property.Builder().buildPartial();
+  private final NullablePropertyFactory factory = new NullablePropertyFactory();
+
+  @Before
+  public void setup() {
+    when(config.getProperty()).thenReturn(property);
+  }
+
+  @Test
+  public void notNullable() {
+    ExecutableElement getterMethod = (ExecutableElement) model.newElementWithMarker(
+        "package com.example;",
+        "public class DataType {",
+        "  ---> public abstract String getName();",
+        "  public static class Builder extends DataType_Builder {}",
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The String literal "  public static class Builder extends DataType_Builder {}" appears 4 times in this file; the first occurrence is on line 61](https://www.codacy.com/app/Chris-Purcell-39/FreeBuilder/file/2277828624/issues/source?bid=3146243&fileBranchId=3187161#l61)